internal/common/settings.go

```diff
@@ -132,3 +137,20 @@ func readFileSourceOfTrustSettings() (*FileTrustSourceSettings, error) {
 
 	return fileSettings, nil
 }
+
+func readSpireSourceOfTrustSettings() (*SpireTrustSourceSettings, error) {
+	if !viper.IsSet("trustsource.spire.domains") {
+		return nil, errors.New("trustsource.spire.domains is required but not found")
+	}
+
+	spireSettings := new(SpireTrustSourceSettings)
```

Comment:
Why use `new()` here? 

Why not just do this?
```suggestion
      spireSettings := &SpireTrustSourceSettings{
        SpireEndpoints: viper.GetStringMapString("trustsource.spire.domains"),
        CertStorePath: viper.GetString("trustsource.spire.certLocation"),
      }
```